### PR TITLE
✨ Allow friends group participant deletion

### PR DESCRIPTION
**Pull request by bjlaa**

### CODE REVIEW

File: src/app/(simulation)/(large-layout)/amis/invitation/_components/LaconicRanking.tsx [hunk 1]
```diff
@@ -25,12 +25,15 @@ export default function LaconicRanking({ group }: Props) {
   }
 
   const particpantsOrdered = group.participants.sort((a, b) => {
-    if (!a.simulation.computedResults || !b.simulation.computedResults) {
+    const computedResultsA = a.simulation?.computedResults
+    const computedResultsB = b.simulation?.computedResults
```
Comment (paulsouche) [inline src/app/(simulation)/(large-layout)/amis/invitation/_components/LaconicRanking.tsx:29]: It was not optional before ?

File: src/app/(simulation)/(large-layout)/amis/resultats/_components/GroupResults.tsx [hunk 2]
```diff
@@ -9,16 +9,20 @@ import { useGetGroupStats } from '@/hooks/groups/useGetGroupStats'
...
 
-type Props = {
+export default function GroupResults({
+  group,
+  refetchGroup,
+}: {
   group: Group
-}
-export default function GroupResults({ group }: Props) {
+  refetchGroup: () => Promise<QueryObserverResult<Group, Error>>
```
Comment (paulsouche) [inline src/app/(simulation)/(large-layout)/amis/resultats/_components/GroupResults.tsx:24]: Too bad remove participant is not PUT group... But anyway, that's alright

File: src/app/(simulation)/(large-layout)/amis/resultats/_components/groupResults/Classement.tsx [hunk 3]
```diff
@@ -55,12 +63,13 @@ export default function Classement({ group }: { group: Group }) {
           }
 
           const { formattedValue, unit } = formatCarbonFootprint(
-            participant?.simulation?.computedResults[defaultMetric].bilan ?? ''
+            participant?.simulation?.computedResults?.[defaultMetric]?.bilan ??
```
Comment (paulsouche) [inline src/app/(simulation)/(large-layout)/amis/resultats/_components/groupResults/Classement.tsx:66]: Why adding this optional marks (more below)

File: src/app/(simulation)/(large-layout)/amis/resultats/_components/groupResults/classement/ClassementMember.tsx [hunk 4]
```diff
@@ -1,40 +1,133 @@
...
+              />
+            </button>
+
+            {isConfirmationModalOpen && (
+              <ConfirmationModal
+                onConfirm={handleDelete}
+                closeModal={() => setIsConfirmationModalOpen(false)}>
+                <p className="text-sm md:text-base">
+                  <Trans>
+                    Supprimer ce participant ? Cette opération est définitive.
```
Comment (paulsouche) [inline src/app/(simulation)/(large-layout)/amis/resultats/_components/groupResults/classement/ClassementMember.tsx:124]: Add translation I guess

> définitive

A bit weird you have to invite him again I guess

File: src/app/(simulation)/(large-layout)/amis/invitation/_components/LaconicRanking.tsx (hunk 1 above)
Comment (bjlaa) [reply src/app/(simulation)/(large-layout)/amis/invitation/_components/LaconicRanking.tsx:29]: true

File: src/app/(simulation)/(large-layout)/amis/resultats/_components/GroupResults.tsx (hunk 2 above)
Comment (bjlaa) [reply src/app/(simulation)/(large-layout)/amis/resultats/_components/GroupResults.tsx:24]: Hm yes we need to be more cautious about using the right request type. I tend to use POST in 80% of cases

File: src/app/(simulation)/(large-layout)/amis/resultats/_components/groupResults/Classement.tsx (hunk 3 above)
Comment (bjlaa) [reply src/app/(simulation)/(large-layout)/amis/resultats/_components/groupResults/Classement.tsx:66]: Removed it, no lint errors

File: src/app/(simulation)/(large-layout)/amis/resultats/_components/groupResults/classement/ClassementMember.tsx (hunk 4 above)
Comment (bjlaa) [reply src/app/(simulation)/(large-layout)/amis/resultats/_components/groupResults/classement/ClassementMember.tsx:124]: Seems to me that in this case that the word "définitive" relatively clearly means there is not way to undo the action but I'll take any other suggestions !

**Review by paulsouche — APPROVED** [review]

**Pull request merged**